paths.go

```diff
@@ -0,0 +1,128 @@
+package quamina
+
+import (
+	"fmt"
+	"strings"
+)
+
+const PathSeparator = "\n"
+
+type pathsIndex struct {
+	root bool
+
+	nodes map[string]*pathsIndex
+
+	// fields map from it's name to it's full path (as specific in quamina.Field)
+	// will be present only on the leafs
+	fields map[string][]byte
+}
+
+func newPathsIndex() *pathsIndex {
```

Comment:
maybe `newRootIndex` or some such, since it's not general-purpose but only used for root node?

Reply:
Actually it's being used in `copy` on all nodes both root and non-root, I'll refactor it so I'll have two methods one for root and another for nodes.

Reply:
Refactored this method to two methods.